Sources/AppStoreConnectCLI/Services/Operations/CreateBetaGroupOperation.swift

```diff
@@ -10,53 +10,41 @@ struct CreateBetaGroupOperation: APIOperation {
         let createBetaGroup: (APIEndpoint<BetaGroupResponse>) -> Future<BetaGroupResponse, Error>
     }
 
-    private let options: CreateBetaGroupOptions
     private let getAppsOperation: GetAppsOperation
+    private let createBetaGroupEndpoint: (_ appId: String) -> APIEndpoint<BetaGroupResponse>
 
     init(options: CreateBetaGroupOptions) {
-        self.options = options
-        self.getAppsOperation = GetAppsOperation(options: .init(bundleIds: [options.appBundleId]))
+        getAppsOperation = GetAppsOperation(options: .init(bundleIds: [options.appBundleId]))
+        createBetaGroupEndpoint = { appId in
+            .create(
+                betaGroupForAppWithId: appId,
+                name: options.groupName,
+                publicLinkEnabled: options.publicLinkEnabled,
+                publicLinkLimit: options.publicLinkLimit,
+                publicLinkLimitEnabled: options.publicLinkLimit != nil
+            )
+        }
     }
 
     private typealias AppAttributes = AppStoreConnect_Swift_SDK.App.Attributes
     private typealias BetaGroupAttributes = AppStoreConnect_Swift_SDK.BetaGroup.Attributes
 
     func execute(with dependencies: CreateBetaGroupDependencies) -> AnyPublisher<BetaGroup, Error> {
-        let options = self.options
-
         let app = getAppsOperation
             .execute(with: .init(apps: dependencies.apps))
```

Comment:
🤔 I find this type inference with `.init` a bit confusing. Not clear what is being provided to the method. But maybe that is just because I'm unfamiliar with the code and don't have IDE conveniences in GitHub pull requests.

Reply:
Yeah I agree `init` can be a little obtuse sometimes. I sometimes gravitate towards it if I'm inlining the initailiser but explicit types increase readability